### PR TITLE
[Composable] Add more sharding strategies to runtime test

### PR DESCRIPTION
**Pull request by rohan-varma**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #106003
* #106002
* #106001
* #106000
* #105999
* #105995
* #105991
* #105207
* #105206
* __->__ #105205
* #105180
* #105176

Add more sharding strategies to ensure equivalence

Differential Revision: [D47462392](https://our.internmc.facebook.com/intern/diff/D47462392/)

### CODE REVIEW

File: test/distributed/_composable/fully_shard/test_fully_shard_runtime.py [hunk 1]
```diff
@@ -92,19 +95,24 @@ def test_training(self):
                 "fsdp_wrap_mode": [
                     FSDPWrapMode.AUTO_WRAP,
                     FSDPWrapMode.MANUAL_WRAP,
+                ],
+                "sharding_strategy": [
+                    ShardingStrategy.SHARD_GRAD_OP,
```
Comment (awgu) [inline test/distributed/_composable/fully_shard/test_fully_shard_runtime.py:100]: I think?
```suggestion
                    ShardingStrategy.FULL_SHARD,
```

**Review by awgu — APPROVED** [review]
LGTM. (check lint)

**Comment by rohan-varma** [comment]
fake tensor failure https://github.com/pytorch/pytorch/actions/runs/5663535346/job/15345654211?pr=105205 issue is a KP and fixed on master.

**Pull request closed without merging**